ppd/testppdfile.c

```diff
@@ -166,7 +166,7 @@ int main(int argc,	    // I - Number of command-line args
   len_output = cupsArrayCount(output);
   
   (cupsArrayFirst(output));
-  for (int j = 1; j< len_output; j++)
+  for (int j = 0; j< len_output-1; j++)
```

Comment:
Also wrong. It must be
```
for (int j = 0; j < len_output; j++)
```

Reply:
The len_output includes the initialization value of " " for the array, which we don't want to be printed in the output screen, so I am using len_output-1 times looping so that " " is not printed in output screen.

Reply:
You do not need to add a "` `" line to the array to initialize it. The array is already initialized and accepts regular lines right after creating it with `cupsArrayNew3()`, when it is still empty. So do not add this empty line at the furst place.

Reply:
But the desired output does not come when not using " " to initialize the array.

Reply:
Now, I have removed the " " line. However, j will loop till len_output-1 as I am using cupsArrayNext as well within the loop, so in I use len_output instead, the length of array will be exceeded when calling cupsArrayNext and will give errors.

Reply:
For outputting all element of a CUPS array I even do not define an integer variable to count the element. I simply do
```
char *line;
for (line = (char *)cupsArrayFirst(output);
     line;
     line = (char *)cupsArrayNext(output))
  puts(output);
```
This outputs all the elements of the array, without needing to get the number of elements and without treating the first element differently. You find this way to treat a CUPS array at many places in CUPS, libcupsfilters, and other OpenPrinting projects.